src/core/ucc_context.c

```diff
@@ -884,6 +885,31 @@ ucc_status_t ucc_context_get_attr(ucc_context_t      *context,
         context_attr->ctx_addr = context->attr.ctx_addr;
     }
 
+    if (context_attr->mask & UCC_CONTEXT_ATTR_FIELD_WORK_BUFFER_SIZE) {
+        int                 i;
+        ucc_base_ctx_attr_t attr;
+        ucc_tl_lib_t *      tl_lib;
+
+        attr.attr.mask = UCC_CONTEXT_ATTR_FIELD_WORK_BUFFER_SIZE;
+        attr.attr.global_work_buffer_size = 0;
+        for (i = 0; i < context->n_tl_ctx; i++) {
+            tl_lib =
+                ucc_derived_of(context->tl_ctx[i]->super.lib, ucc_tl_lib_t);
+            status = tl_lib->iface->context.get_attr(&context->tl_ctx[i]->super,
+                                                     &attr);
+            if (UCC_OK != status) {
+                ucc_error("failed to obtain global work buffer size");
+                return status;
+            }
+            /* currently only TL/UCP will return a value */
```

Comment:
lets not make such assumption. just find MAX over all TLs ? Other TLs should return 0, so result will still be TL/UCP buffer size